solutions/src/test/java/ru/sberbank/school/task02/ReverseCalculatorTest.java

```diff
@@ -0,0 +1,200 @@
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.001).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    void convertReversedDeltaBuy() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB,
+                BigDecimal.valueOf(50_000), 20_000, Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.0005).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    void convertReversedHugeDeltaBuy() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Optional<BigDecimal> optionalOffer = calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB,
+                BigDecimal.valueOf(275_000), 150_000, Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.00025).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    void convertReversedNullOperationThrowsNPeWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Assertions.assertThrows(NullPointerException.class, () -> {
+            calculator.convertReversed(null, Symbol.USD_RUB, BigDecimal.valueOf(1), Beneficiary.CLIENT);
+        }, "No operation provided");
+    }
+
+    @Test
+    void convertReversedNullSymbolThrowsNPeWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Assertions.assertThrows(NullPointerException.class, () -> {
+            calculator.convertReversed(ClientOperation.BUY, null, BigDecimal.valueOf(1), Beneficiary.CLIENT);
+        }, "No symbol provided");
+    }
+
+    @Test
+    void convertReversedNullAmountThrowsNPeWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Assertions.assertThrows(NullPointerException.class, () -> {
+            calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, null, Beneficiary.CLIENT);
+        }, "No amount provided");
+    }
+
+    @Test
+    void convertReversedZeroAmountThrowsIAe() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, BigDecimal.ZERO, Beneficiary.CLIENT);
+        }, "Wrong amount");
+}
```

Comment:
&#39;method def rcurly&#39; has incorrect indentation level 0, expected level should be 4.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1218725811>*